src/jdk/src/share/sample/jmx/heapmemorymonitor/HeapMemoryMonitor.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2019, Amazon.com and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryUsage;
+import java.lang.management.MemoryType;
+import java.lang.String;
+import java.lang.System;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+
+// Run with a heap size tailored to HeapMemoryMonitor. allocateObjects().
+// -Xmx50m works fine with the defaults.
+
+class HeapMemoryMonitor {
+
+    private long usedKb = 0;
+    private long totalKb = 0;
+
+    private long afterGCUsedKb = 0;
+    private long afterGCTotalKb = 0;
+
+    private final List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
+    private final List<MemoryPoolMXBean> heapMemoryPoolMXBeans = new ArrayList<MemoryPoolMXBean>();
+    private final List<MemoryPoolMXBean> afterGCHeapMemoryPoolMXBeans = new ArrayList<MemoryPoolMXBean>();
+
+    private final DecimalFormat df = new DecimalFormat("###");
+
+    public static void main(String[] args) throws Exception {
+        HeapMemoryMonitor h = new HeapMemoryMonitor();
+        h.run();
+    }
+
+    public HeapMemoryMonitor() {
+        System.out.println("Monitor heap memory utilization.");
+
+	String memoryPoolName;
+
+        for (MemoryPoolMXBean memoryPool : memoryPoolMXBeans) {
+	    //	    System.out.println(memoryPool.getName() + ' ' + memoryPool.getType());
+	    if (memoryPool.getType() != MemoryType.HEAP) {
+                // Ignore non-HEAP memory pools
+                continue;
+            }
+            heapMemoryPoolMXBeans.add(memoryPool);
+
+            memoryPoolName = memoryPool.getName();
```

Comment:
This variable would better be declared here, and final. Better yet: use the expression and have no variable, since the call memoryPool,getName() is sufficiently descriptive already.